packages/@jsii/dotnet-runtime/src/Amazon.JSII.Analyzers.UnitTests/Helpers/DiagnosticResult.cs

```diff
@@ -54,10 +89,50 @@ public DiagnosticResultLocation[] Locations
 
         public string Message { get; set; }
 
-        public string Path => Locations.Length > 0 ? Locations[0].Path : "";
+        public string Path => Locations.Count > 0 ? Locations[0].Path : "";
+
+        public int Line => Locations.Count > 0 ? Locations[0].Line : -1;
 
-        public int Line => Locations.Length > 0 ? Locations[0].Line : -1;
+        public int Column => Locations.Count > 0 ? Locations[0].Column : -1;
 
-        public int Column => Locations.Length > 0 ? Locations[0].Column : -1;
+        public override bool Equals(object? obj)
+        {
+            if (obj is DiagnosticResult other)
+            {
+                return Equals(other);
+            }
+
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            return Id.GetHashCode()
+                   ^ Severity.GetHashCode()
+                   ^ Message.GetHashCode()
+                   ^ Path.GetHashCode()
+                   ^ Line.GetHashCode()
+                   ^ Column.GetHashCode();
+        }
+
+        public bool Equals(DiagnosticResult other)
+        {
+            return Id == other.Id
+                   && Severity == other.Severity
+                   && Message == other.Message
+                   && Path == other.Path
+                   && Line == other.Line
+                   && Column == other.Column;
```

Comment:
This is making the assumption that two `DiagnosticResults` with the same first `DiagnosticResultLocation` are the same, regardless of the other `DiagnosticResultLocation` in `Locations`. I lack context to know if that's valid or not; I _suspect_ it's ok, but worth calling it out.